cp1616/src/cp1616_io_controller.cpp

```diff
@@ -448,6 +467,47 @@ void Cp1616IOController::setSemModChange(int mod_change)
   sem_mod_change_ = mod_change;
 }
 
+PNIO_UINT32 Cp1616IOController::parseConfigFile(std::string filepath)
+{
+  std::ifstream fin(filepath.c_str());
+  YAML::Parser parser(fin);
+  YAML::Node doc;
+  PNIO_UINT32 error_code = PNIO_OK;
+  
+  try
+  {
+    parser.GetNextDocument(doc);
+      
+    for(unsigned i = 0; i < doc.size(); i++)
+    {
+      ControllerModuleData temp_module;
+      doc[i] >> temp_module; 
+      if(temp_module.type == "input")
```

Comment:
Not really, although the static variables can be defined at the top of the cpp in order to avoid polluting the header.  Using "hard coded" keys in the `>>` operator is fine because they are aren't buried.  These values are buried in the code.  If you decided to change them, it would be difficult to find them.